relion/_parser/ctffind.py

```diff
@@ -1,7 +1,24 @@
+from gemmi import cif
+from pathlib import Path
+import os
+import functools
+from collections import namedtuple
+
+CTFMicrograph = namedtuple(
+    "CTFMicrograph", ["total_motion", "early_motion", "late_motion"]
+)
+
+
 class CTFFind:
     def __init__(self, path):
         self._basepath = path
         self._jobcache = {}
+        self.val_astigmatism = None
+        self.val_defocus_u = None
+        self.val_defocus_v = None
+        self.val_defocus_angle = None
+        self.val_max_resolution = None
+        self.val_fig_or_merit = None
```

Comment:
I think we don't need those attributes. Instead of having for each value
* an attribute,
* a property to read it, and
* a set_attribute function to set it by calling a `find_values` function

we should be able to get away with simply
* a property to read it, which calls `self._find_values` to read it (note the leading `_`, I don't think this should be a public function)

Reply:
Okay this makes sense, will implement